components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/attest_iat_claims_loader.c

```diff
@@ -0,0 +1,228 @@
+{
+    uint32_t lc = psa_security_lifecycle_state();
+    switch (lc) {
+        case PSA_LIFECYCLE_UNKNOWN:
+            return TFM_SLC_UNKNOWN;
+        case PSA_LIFECYCLE_ASSEMBLY_AND_TEST:
+            return TFM_SLC_ASSEMBLY_AND_TEST;
+        case PSA_LIFECYCLE_PSA_ROT_PROVISIONING:
+            return TFM_SLC_PSA_ROT_PROVISIONING;
+        case PSA_LIFECYCLE_SECURED:
+            return TFM_SLC_SECURED;
+        case PSA_LIFECYCLE_NON_PSA_ROT_DEBUG:
+            return TFM_SLC_NON_PSA_ROT_DEBUG;
+        case PSA_LIFECYCLE_RECOVERABLE_PSA_ROT_DEBUG:
+            return TFM_SLC_RECOVERABLE_PSA_ROT_DEBUG;
+        case PSA_LIFECYCLE_DECOMMISSIONED:
+            return TFM_SLC_DECOMMISSIONED;
+        default:
+            return TFM_SLC_UNKNOWN;
+    }
+}
+
+/* Hash of attestation public key */
+static enum tfm_plat_err_t attest_public_key_sha256(uint32_t *size, uint8_t *buf)
+{
+    const psa_key_id_t key_id = PSA_ATTESTATION_PRIVATE_KEY_ID;
+    psa_key_handle_t handle = 0;
+
+    uint8_t *public_key = NULL;
+    psa_key_type_t type;
+    psa_key_type_t public_type;
+    size_t bits;
+    size_t public_key_size = 0;
+    size_t public_key_length = 0;
+
+    psa_status_t crypto_ret;
+    psa_hash_operation_t hash_handle;
+
+    crypto_ret = psa_crypto_init();
+    if (crypto_ret != PSA_SUCCESS) {
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    crypto_ret = psa_open_key(PSA_KEY_LIFETIME_PERSISTENT, key_id, &handle);
+    if (crypto_ret != PSA_SUCCESS) {
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    crypto_ret = psa_get_key_information(handle, &type, &bits);
+    if (crypto_ret != PSA_SUCCESS) {
+        psa_close_key(handle);
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+    if (!PSA_KEY_TYPE_IS_ECC(type)) {
+        psa_close_key(handle);
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+    public_type = PSA_KEY_TYPE_PUBLIC_KEY_OF_KEYPAIR(type);
+    public_key_size = PSA_KEY_EXPORT_MAX_SIZE(public_type, bits);
+    public_key = (uint8_t *) malloc(public_key_size);
+    if (public_key == NULL) {
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    crypto_ret = psa_export_public_key(handle,
+                                       public_key, public_key_size,
+                                       &public_key_length);
+    if (crypto_ret != PSA_SUCCESS) {
+        free(public_key);
+        psa_close_key(handle);
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    crypto_ret = psa_hash_setup(&hash_handle, PSA_ALG_SHA_256);
+    if (crypto_ret != PSA_SUCCESS) {
+        free(public_key);
+        psa_close_key(handle);
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    psa_hash_update(&hash_handle, public_key, public_key_length);
+
+    crypto_ret = psa_hash_finish(&hash_handle,
+                                 buf,
+                                 *size,
+                                 (size_t *) size);
+    if (crypto_ret != PSA_SUCCESS) {
+        free(public_key);
```

Comment:
too many same kind of operations: free, close
please consider goto with appropriate error code